cw_core/lib/utils/socket_health_logger.dart

```diff
@@ -0,0 +1,154 @@
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+import 'package:cw_core/utils/print_verbose.dart';
+import 'package:cw_core/wallet_type.dart';
+import 'package:path_provider/path_provider.dart';
+
+class SocketHealthLogEntry {
+  final DateTime timestamp;
+  final WalletType walletType;
+  final String walletName;
+  final bool isHealthy;
+  final String? error;
+  final String? syncStatus;
+  final bool wasReconnected;
+  final String trigger;
+
+  SocketHealthLogEntry({
+    required this.timestamp,
+    required this.walletType,
+    required this.walletName,
+    required this.isHealthy,
+    this.error,
+    this.syncStatus,
+    required this.wasReconnected,
+    required this.trigger,
+  });
+
+  String toLogString() {
+    final dateStr = timestamp.toIso8601String();
+    final status = isHealthy ? 'HEALTHY' : 'UNHEALTHY';
+    final reconnect = wasReconnected ? 'RECONNECTED' : 'NO_RECONNECT';
+    final errorStr = error != null ? ' | Error: $error' : '';
+    final syncStr = syncStatus != null ? ' | Sync: $syncStatus' : '';
+
+    return '[$dateStr] $trigger | ${walletType.name} | $walletName | $status | $reconnect$errorStr$syncStr';
+  }
+
+  Map<String, dynamic> toJson() {
+    return {
+      'timestamp': timestamp.toIso8601String(),
+      'walletType': walletType.name,
+      'walletName': walletName,
+      'isHealthy': isHealthy,
+      'error': error,
+      'syncStatus': syncStatus,
+      'wasReconnected': wasReconnected,
+      'trigger': trigger,
+    };
+  }
+
+  factory SocketHealthLogEntry.fromJson(Map<String, dynamic> json) {
+    return SocketHealthLogEntry(
+      timestamp: DateTime.parse(json['timestamp'] as String),
+      walletType: WalletType.values.firstWhere((e) => e.name == json['walletType'] as String),
+      walletName: json['walletName'] as String,
+      isHealthy: json['isHealthy'] as bool,
+      error: json['error'] as String?,
+      syncStatus: json['syncStatus'] as String?,
+      wasReconnected: json['wasReconnected'] as bool,
+      trigger: json['trigger'] as String,
+    );
+  }
+}
```

Comment:
this is an unversioned struct (which is a non-issue if it is RAM-only class), if we decide to store it to disk we probably should add a version and be much more forgiving in term of types, for example:

`as String` -> `?.toString()`
`as String?` -> `?.toString() ?? ""`
`as bool` -> `?.toString == "true"`
`WalletType.values.firstWhere((e) => e.name == json['walletType'] as String)` -> I don't have an oneliner but switch statement would be much more efficient here - or even better - just store index of the enum

Reply:
Yep, makes sense

Removed this though, as we no more need it